### PR TITLE
Improve the packing of RefPosition

### PR DESCRIPTION
**Pull request by BruceForstall**
The LSRA RefPosition type was not packing well. In particular,
bitfields don't pack unless they have the same base type.

Move all the data member declarations to the top, and make them adjacent,
so it is easier to see them all together. Also, change all bitfields
to "unsigned char" base type.

This reduces the size from 36 to 28 bytes for x86, and 48 to 40 bytes
for x64 (release, on Windows).

### CODE REVIEW

**Comment by BruceForstall** [comment]
@CarolEidt PTAL
cc @dotnet/jit-contrib

File: src/jit/lsra.h [hunk 1]
```diff
@@ -1445,6 +1445,106 @@ class Interval : public Referenceable
 class RefPosition
 {
 public:
+
+    // A RefPosition refers to either an Interval or a RegRecord. 'referent' points to one
+    // of these types. If it refers to a RegRecord, then 'isPhysRegRef' is true. If it
+    // refers to an Interval, then 'isPhysRegRef' is false.
+    //
+    // Q: can 'referent' be NULL?
```
Comment (CarolEidt) [inline src/jit/lsra.h]: I don't believe that `referent` can ever by null.

**Comment by BruceForstall** [comment]
Failure in `Windows_NT x64 Formatting`:
```
21:02:13 D:\j\workspace\x64_windows_n---616f30f2>python -u tests\scripts\format.py -c D:\j\workspace\x64_windows_n---616f30f2 -o Windows_NT -a x64 
21:02:13 Downloading .Net CLI
21:02:35 Traceback (most recent call last):
21:02:35   File "tests\scripts\format.py", line 248, in <module>
21:02:35     return_code = main(sys.argv[1:])
21:02:35   File "tests\scripts\format.py", line 102, in main
21:02:35     testfile.retrieve(request_url, dotnetcliFilename)
21:02:35   File "C:\Python27\lib\urllib.py", line 273, in retrieve
21:02:35     block = fp.read(bs)
21:02:35   File "C:\Python27\lib\socket.py", line 384, in read
21:02:35     data = self._sock.recv(left)
21:02:35   File "C:\Python27\lib\ssl.py", line 766, in recv
21:02:35     return self.read(buflen)
21:02:35   File "C:\Python27\lib\ssl.py", line 653, in read
21:02:35     v = self._sslobj.read(len)
21:02:35 socket.error: [Errno 10054] An existing connection was forcibly closed by the remote host
21:02:35 
```
@adiaaida Ever seen this?

https://ci.dot.net/job/dotnet_coreclr/job/master/job/x64_windows_nt_formatting_prtest/7671/

**Comment by BruceForstall** [comment]
@dotnet-bot test Windows_NT x64 Formatting
@dotnet-bot test Ubuntu arm Cross Release Build

**Pull request merged**